src/monster.cpp

```diff
@@ -2250,8 +2250,27 @@ void monster::drop_items_on_death()
     if( type->death_drops.empty() ) {
         return;
     }
-    const auto dropped = g->m.put_items_from_loc( type->death_drops, pos(),
-                         calendar::start_of_cataclysm );
+
+    std::vector<item> items = item_group::items_from( type->death_drops, calendar::start_of_cataclysm );
+
+    // This block removes some items, according to item spawn scaling factor
+    const float spawn_rate = get_option<float>( "ITEM_SPAWNRATE" );
+    if( spawn_rate < 1 ) {
+        // Temporary vector, to remember which items will be dropped
+        std::vector<item> remaining;
+        for( item it : items ) {
```

Comment:
```suggestion
        for( const item &it : items ) {
```

Reply:
I understand why `const` is good practice here, but would you mind explaining why we want to use an address here? (sorry, bit of a c++ noob)

Reply:
If it captures a reference, it does not make a copy. The item class is large enough that it's imperative to avoid making copies unecesssrally.

Reply:
Oh, holy crap! I didn't know it made a copy! Yeah, I definitely see why that's better now. Thanks.